AEPServices/Sources/ui/GlobalUIMessaging.swift

```diff
@@ -0,0 +1,31 @@
+/*
+ Copyright 2020 Adobe. All rights reserved.
+ This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License. You may obtain a copy
+ of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software distributed under
+ the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ OF ANY KIND, either express or implied. See the License for the specific language
+ governing permissions and limitations under the License.
+ */
+
+import Foundation
+
+/// Global UI Message delegate which is used to listen for current message lifecycle events
+@objc(AEPGlobalUIMessaging) public protocol GlobalUIMessaging {
```

Comment:
In the doc comments, I see the mention of `message`, but don't see them in the function definition.

I'd be in favor of having all 3 protocol functions pass in a `message`, otherwise, it could be difficult to gain context into which messages are showing, dismissing, etc.

Reply:
We have this global listener as a common listener for end user so they can listen for the message monitor activities like show and dismiss. As this is a common method for all kind of message. 
I don't think message object is useful.

Also if we want to give a message back then we will need a common base class for all kind of message. But that base class might not have anything in common other than being a type for messages.